### PR TITLE
Do not track testing coverage for styled.js* files

### PR DESCRIPTION
**Pull request by gusaiani**
Excludes .styled.jsx (and .styled.js for now, we have not yet converged on just one suffix) from coverage reports.

### How to test

Run tests: `yarn test-unit --coverage` 

Open lcov report: `open coverage/lcov-report/index.html`

Go to coverage for any directory that includes a styled component, for example:

http://……/frontend/src/metabase/components/DrawerSection/index.html

Now there should be no styled.js{x} file in the coverage report.

### CODE REVIEW

**Comment by ariya** [comment]
This seems to be a bit complicated. How about something like this?

```diff
--- a/jest.unit.conf.json
+++ b/jest.unit.conf.json
@@ -28,7 +28,8 @@
     "frontend/src/**/*.js",
     "frontend/src/**/*.jsx",
     "enterprise/frontend/src/**/*.js",
-    "enterprise/frontend/src/**/*.jsx"
+    "enterprise/frontend/src/**/*.jsx",
+    "!**/*.styled.{js,jsx}"
   ],
   "coveragePathIgnorePatterns": [
     "/node_modules/",
```

**Comment by gusaiani** [comment]
> This seems to be a bit complicated. How about something like this?

@ariya The JS question: what do you mean by **this**? 
Because I was experimenting and quickly force-pushing the attempts.

**Comment by gusaiani** [comment]
@ariya Also, ideally we'd just converge on either `.styled.js` or `.styled.jsx`...and this would get further simplified.

**Comment by ariya** [comment]
> @ariya Also, ideally we'd just converge on either `.styled.js` or `.styled.jsx`...and this would get further simplified.

Sure, but let's get this in first (we already have many files with either extension and tackle the renaming separately.

**Comment by gusaiani** [comment]
@ariya 

> Sure, but let's get this in first.

Great. 

Did you see the latest iteration of this PR?

As I see it, adding entries to the array `coveragePathIgnorePatterns` should make it clearer to readers that this is to be ignored. 

Wdyt?

File: jest.unit.conf.json [hunk 1]
```diff
@@ -4,7 +4,10 @@
     "\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/frontend/test/__mocks__/fileMock.js",
     "^promise-loader\\?global\\!metabase\\/lib\\/ga-metadata$": "<rootDir>/frontend/src/metabase/lib/ga-metadata.js"
   },
-  "testPathIgnorePatterns": ["<rootDir>/frontend/test/.*/.*.tz.unit.spec.js"],
+  "testPathIgnorePatterns": [
+    "<rootDir>/frontend/test/.*/.*.tz.unit.spec.js",
+    "<rootDir>/frontend/test/.*/.*.styled.js*"
```
Comment (ariya) [inline jest.unit.conf.json:9]: Isn't this to specify the test files and not coverage?

Comment (gusaiani) [reply jest.unit.conf.json:9]: Coverage — yes, the draft PR title was misleading, apologies for that.

Comment (ariya) [reply jest.unit.conf.json:9]: I still don't follow, is it necessary to add a new entry to `testPathIgnorePatterns`?

Comment (gusaiani) [reply jest.unit.conf.json]: No, not necessary at all, my mixup from inattentive experimentation.

Removed.

**Review by ariya — APPROVED** [review]
LGTM :+1:  Thanks for taking care of this!

**Pull request merged**